patches/api/0426-PreSpawnerSearchEvent.patch

```diff
@@ -0,0 +1,95 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TreemanKing <67459602+TreemanKing@users.noreply.github.com>
+Date: Wed, 16 Aug 2023 11:59:08 +1000
+Subject: [PATCH] PreSpawnerSearchEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/PreSpawnerSearchEvent.java b/src/main/java/io/papermc/paper/event/entity/PreSpawnerSearchEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..b48b5208e04a920e49a984addb7ee7093509ba48
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/PreSpawnerSearchEvent.java
+@@ -0,0 +1,83 @@
++package io.papermc.paper.event.entity;
++
++import com.google.common.base.Preconditions;
++import org.bukkit.event.Cancellable;
++import org.bukkit.Location;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Event;
++import org.bukkit.event.HandlerList;
++import org.jetbrains.annotations.ApiStatus;
++import org.jetbrains.annotations.NotNull;
++import java.util.Collection;
++import java.util.Collections;
++
++public class PreSpawnerSearchEvent extends Event implements Cancellable {
++    @NotNull private final Location spawnerLocation;
++    @NotNull private final Collection<Player> activatingPlayers;
++    private int backoffTicks;
++    private boolean cancelled = false;
++    private static final HandlerList HANDLER_LIST = new HandlerList();
++
++    @ApiStatus.Internal
++    public PreSpawnerSearchEvent(@NotNull Location spawnerLocation, @NotNull Collection<Player> activatingPlayers, int backoffTicks) {
++        this.spawnerLocation = Preconditions.checkNotNull(spawnerLocation, "Spawner Location can not be null");
++        this.activatingPlayers = Preconditions.checkNotNull(activatingPlayers, "There must be a activating player");
++        this.backoffTicks = backoffTicks;
++    }
++
++    /**
++     * @return The spawner's location
++     */
++    @NotNull
++    public Location getSpawnerLocation() {
++        return spawnerLocation;
++    }
++
++    /**
++     * @return A collection of players who are close enough to activate the spawner
```

Comment:
Should denote a) via annotation and b) in the javadocs that the returned collection is immutable.